BacktraceScreen.c

```diff
@@ -0,0 +1,301 @@
+#include "BacktraceScreen.h"
+
+#include <execinfo.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <ctype.h>
+#include <unistd.h>
+
+#include "FunctionBar.h"
+#include "InfoScreen.h"
+#include "Panel.h"
+#include "Object.h"
+#include "Vector.h"
+#include "XUtils.h"
+#include "sys/wait.h"
+
+#define DWARF_INFORMATION_NOT_FOUND 1
+
+static const char* const BackTraceScreenFunctions[] = {"Refresh", "Done", NULL};
+static const char* const BackTraceScreenKeys[] = {"F3", "Esc"};
+static const int  BackTraceScreenEvents[] = {KEY_F(3), 27};
+
+static char *BacktraceScreen_parseEuStackFunction(char *current_line) {
+   while (current_line  && isspace(*current_line)) {current_line++;}
```

Comment:
1. Variable name style. In htop codebase the variables are usually named in camelCase, not snake_case.
2. Is it possible for `current_line` to be NULL? It is confusing when the check for NULL pointer is not written explicitly. So you mean either `current_line != NULL` or `*current_line != '\0'`, which one?

Reply:
The latter in this case, as based on the C standard p++ on a non-NULL (valid) pointer always returns a valid pointer. Thus assuming the compiler would take the code as is, it would be allowed to check the pointer once before the loop and just keep the second test as the loop condition (constant folding + loop invariant optimization). As that's clearly not, what's intended, it must be a check for the end of the string, in which case `*p` is sufficient. And personally I'd even prefer `*p` over `*p != 0` or `*p != '\0'`. DRY. There's quite some code in htop already, that omits the `!= NULL` or `!= 0` in most cases, if the context those checks appear in are clear enough on what that check needs to be. In this example, repeatedly checking on an iterated string to suddenly become NULL makes no sense (actually would imply a bug in itself given the C standard), thus the end of string check is the most sensible choice; which is supported by the actual structure of that loop, which just increments it, instead of e.g. following some random links of a linked list or such.

BTW: That was also, why I noticed that bug in my review: That loop without the de-reference made little sense and looked "strange" based on experience.

Reply:
I need to change my habits concerning the rule of naming variables 😄